### PR TITLE
VMware: New module vmware_host_snmp

### PR DESCRIPTION
**Pull request by ckotte**
##### SUMMARY
New module to configure SNMP agent on an ESXi host.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
vmware_host_snmp

##### ANSIBLE VERSION
```
2.8-devel
```

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+        - The embedded SNMP agent receives hardware events either from IPMI sensors C(sensors) or CIM indications C(indications).
+    type: str
+    choices: [ indications, sensors ]
+    default: 'indications'
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Enable and configure SNMP
+  vmware_host_ad_auth:
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]:
```suggestion
  vmware_host_snmp:
```

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    domain: example.local
+    join_user: adjoin
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: I think these parameters are not related to this module.

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+    join_password: Password123$
+    state: enabled
+    validate_certs: no
+  delegate_to: localhost
+
+- name: Disable SNMP
+  vmware_host_ad_auth:
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: Same here.

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+RETURN = r'''
+results:
+    description: metadata about host system's SNMP configuration
+    returned: always
+    type: dict
+    sample: {
+        "esxi01": {
+            "changed": true,
+            "domain": "example.local",
+            "membership_state": "ok",
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: I think this irrelevant, need to change according to SNMP output.

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+                            changed_list.append("HW source")
+                            results['hw_source_previous'] = option.value
+                            option.value = hw_source
+                    # TODO: log level
+                    # loglevel -> info (default)
+            # TODO: trap filter
+            # Comma separated list of trap oids for traps not to be sent by agent
+            # filter: EventFilter -> [1.3.6.1.4.1.6876.4.1.1.0, 1.3.6.1.4.1.6876.4.1.1.1]
+
+            # self.module.fail_json(msg=snmp_config_spec
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: remove this code if it not used.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Does VCSIM support SNMP configuration ?

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py (hunk 1 above)
Comment (ckotte) [reply lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: fixed

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py (hunk 2 above)
Comment (ckotte) [reply lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: fixed

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py (hunk 3 above)
Comment (ckotte) [reply lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: fixed

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py (hunk 4 above)
Comment (ckotte) [reply lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: fixed

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py (hunk 5 above)
Comment (ckotte) [reply lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: fixed

**Comment by ckotte** [comment]
> The test `ansible-test sanity --test docs-build` [[explain](https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/docs-build.html)] failed with 1 error:
> 
> ```
> docs/docsite/rst/user_guide/playbooks_strategies.rst:23:0: duplicate-label: duplicate label strategy_plugins, other instance in /root/ansible/docs/docsite/rst/plugins/strategy.rst
> ```
> [click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

@Akasurde Any idea how to fix this? I didn't change anything in this file...

**Comment by Akasurde** [comment]
@ckotte I guess this is due to e70030961858566592c922435cb6ed5fa3ed7baf

@acozine Can you help us with  ?

**Comment by Akasurde** [comment]
@ckotte There is some error in test case - https://app.shippable.com/github/ansible/ansible/runs/96380/87/console

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+    type: str
+    choices: [ debug, info, warning, error ]
+    default: 'info'
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Enable and configure SNMP community
+  vmware_host_snmp:
+    hostname: '{{ vcenter_hostname }}'
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: If we are using ESXi host as per condition on line number 157, then we should rename this as `{{ esxi_hostname }}`` and respectively other parameters.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Rest LGTM

File: lib/ansible/modules/cloud/vmware/vmware_host_snmp.py (hunk 6 above)
Comment (ckotte) [reply lib/ansible/modules/cloud/vmware/vmware_host_snmp.py]: Yeah, I just copied this part from another module. Fixed. Thx.

**Review by Akasurde — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@pdellaert @dericcrago @Im0 @tchernomax @jeking3 Could you please review this ?
